test/validation/api/time/time.c

```diff
@@ -589,6 +685,95 @@ static void time_test_accuracy_nsec(void)
 	}
 }
 
+static int time_test_global_sync_thr(void *arg ODP_UNUSED)
+{
+	global_shared_mem_t *global_mem;
+	odp_shm_t global_shm;
+	int tid = odp_thread_id();
+
+	global_shm = odp_shm_lookup(GLOBAL_SHM_NAME);
+	global_mem = odp_shm_addr(global_shm);
+
+	if (!global_mem)
+		return 1;
+
+	odp_barrier_wait(&global_mem->test_barrier);
+	global_mem->time[tid][0] = odp_time_global();
+	odp_time_wait_ns(ODP_TIME_MSEC_IN_NS * 100);
+	odp_barrier_wait(&global_mem->test_barrier);
+	global_mem->time[tid][1] = odp_time_global();
+
+	return 0;
+}
+
+static void time_test_global_sync(void)
+{
+	const uint64_t tolerance =
+		odp_cunit_ci() ? TIME_TOLERANCE_CI_NS : TIME_TOLERANCE_NS;
+	int num = global_mem->num_threads;
+	int tid = odp_thread_id();
+
+	/* Test using num threads plus the current thread. */
+	odp_barrier_init(&global_mem->test_barrier, num + 1);
+
+	odp_cpumask_t cpumask;
+	odph_thread_common_param_t thr_common;
+	odph_thread_param_t thr_param;
+	odph_thread_t thread_tbl[MAX_WORKERS];
+
+	odp_cpumask_default_worker(&cpumask, num);
+
+	odph_thread_param_init(&thr_param);
+	thr_param.start = time_test_global_sync_thr;
+	thr_param.thr_type = ODP_THREAD_WORKER;
+
+	odph_thread_common_param_init(&thr_common);
+	thr_common.instance = *instance;
+
+	int i = 0, cpu = odp_cpumask_first(&cpumask);
```

Comment:
At least for me defining variables at the top of a code block makes reading the code easier, although checkpatch doesn't seem to complain about this anymore. `i` is also now defined twice for quite different usage. Perhaps for example `thr` would be better name here.

Reply:
Checkpatch never complained about that, which is good because collecting unrelated declarations in some random place away from where they're used doesn't make any sense. Regardless, I moved some declarations. Variable name changed to "thr". In v6.